### ISSUE TITLE
[suggestion] UI should show that connection is via proxy

### ISSUE THREAD

**Issue by yurivict** [qTox/qTox#3420]
The sentiment that "Tox is unsafe because it exposes your IP" is quite widespread online. So I think that it makes sense to visually indicate that qTox is connected through proxy, therefore hiding your IP. This can be done near the green light indicating the "connected" status, and also in the icon. The letters "PROXY" could be painted over the icon corner.

Of course this whole concept won't be applicable to qTox run in a virtual machine.

**Comment by anoadragon453** [comment]
What do you think of these design concepts?

Icon near username:
![qtox-proxy-next-to-user](https://cloud.githubusercontent.com/assets/1342360/16446070/99ebdc7e-3d9a-11e6-9a42-97d760955a91.png)

Icon over avatar:
![qtox-proxy-over-user](https://cloud.githubusercontent.com/assets/1342360/16446117/d1f15c98-3d9a-11e6-9c73-7604f57baaca.png)

Pattern over availability status:
![qtox-proxy-test](https://cloud.githubusercontent.com/assets/1342360/16446075/9d0fdaa4-3d9a-11e6-90c9-59d14bb097c0.png)

**Comment by yurivict** [comment]
I like the first two.

**Comment by zetok** [comment]
@anoadragon453 Icon tells ~nothing. Same goes for pattern.

If proxy indicator should be available somewhere, why not the program bar?
![bar](https://cloud.githubusercontent.com/assets/3148759/16447498/d8f2f84a-3de4-11e6-821b-0c57acefb5f9.png)
↑ with proxy active, it could have added `- (via SOCKS5 proxy)` / `- (via HTTP proxy)`

**Comment by anoadragon453** [comment]
@zetok I like that idea, saves people a "qtox what is arrow icon" DDG
search. Either `(via SOCKS5 proxy) / - (via HTTP proxy)` or `(SOCKS5
proxy active) / - (HTTP proxy active)` works for me.

I feel the icon would be unnecessary if we have the info in the title.

**Comment by zetok** [comment]
`($type proxy active)` seems to sound better.

**Comment by yurivict** [comment]
Graphical symbols are always easier to read.

**Comment by zetok** [comment]
@yurivict which graphical symbol? The one that has no meaning behind it?

**Comment by yurivict** [comment]
It is true that there is no commonly accepted icon for "proxy" (amazingly). But the blue earth symbol seems to be used a lot. For example something like this http://eproxy.my-addr.com/img/proxy_antivirus_icon.png (w/out the bottom part). Proxy switcher addon in chrome also uses the blue earth symbol.

**Comment by anoadragon453** [comment]
@yurivict We should go for the clearest option to users, which in this case is the text.

The question is whether the text is sufficient to alerting users that there is a proxy active, as a title bar change doesn't really jump out to most.

**Comment by yurivict** [comment]
Exactly, text doesn't jump out as much a graphic representation does.

**Comment by yurivict** [comment]
People can already see that there is the proxy, in Connection Settings. The point is to unambiguously and clearly assure users that their IP address isn't compromised. Otherwise many people just believe that Tox isn't safe and reject it.

**Comment by ElLamparto** [comment]
I would say, the user should be notified when no direct connection is possible (no incoming connections due to a firewall, closed port, etc.)

**Comment by zetok** [comment]
> I would say, the user should be notified when no direct connection is possible (no incoming connections due to a firewall, closed port, etc.)

And how would you know that?

**Comment by ElLamparto** [comment]
@zetok, I do not know, but Bit Torrent applications (and even R.I.P. aMule) do it. Maybe they just wait and display - say - yellow dot until someone connects, and when that happens - a green one?

**Comment by zetok** [comment]
Right. Sorry, my bad, I was asking how that would be done, considering that qTox doesn't get that info from toxcore.

Also, note that it's not possible to tell whether there's no incoming connections because there's no network, or because something was blocked, or because there's literally no DHT network. I.e. qTox is supposed to be functional on even on intranet, with no connection to the internet. In this case there's no way to tell whether there is no connection because there are no (known) peers in the network other than you, or because there's some connection problem.
